### PR TITLE
[IMP] pos_partner_birthdate: Adding birthdate on partner list

### PR DESCRIPTION
**Pull request by etobella**
image

### CODE REVIEW

File: pos_partner_birthdate/static/src/xml/screens.xml
Comment (cvinh) [inline pos_partner_birthdate/static/src/xml/screens.xml:1]: Rename file to screens.xml as declated in manifest

File: pos_partner_birthdate/static/src/js/PosDB.esm.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    _partner_search_string(partner) {
+        var result = super._partner_search_string(partner);
+        if (partner.birthdate_date) {
+            result =
+                result.substr(0, result.length - 1) +
+                "|" +
+                formatDate(DateTime.fromJSDate(new Date(partner.birthdate_date))) +
+                "\n";
+        }
+        console.log(result);
```
Comment (cvinh) [inline pos_partner_birthdate/static/src/js/PosDB.esm.js]: console.log can be removed

**Review by cvinh — CHANGES_REQUESTED** [review]

File: pos_partner_birthdate/static/src/js/PosDB.esm.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const {DateTime} = luxon;
+
+patch(PosDB.prototype, {
+    _partner_search_string(partner) {
+        var result = super._partner_search_string(partner);
+        if (partner.birthdate_date) {
+            result =
+                result.substr(0, result.length - 1) +
+                "|" +
+                formatDate(DateTime.fromJSDate(new Date(partner.birthdate_date))) +
```
Comment (cvinh) [inline pos_partner_birthdate/static/src/js/PosDB.esm.js]: maybe you can improve search to allow user search by date without format...
For example mmddyyyy or mmddyy (en) or ddmmyyyy or ddmmyy (fr)

File: pos_partner_birthdate/static/src/js/PartnerLine.esm.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/** @odoo-module **/
+import {PartnerLine} from "@point_of_sale/app/screens/partner_list/partner_line/partner_line";
+import {formatDate} from "@web/core/l10n/dates";
+import {patch} from "@web/core/utils/patch";
+const {DateTime} = luxon;
+
+patch(PartnerLine.prototype, {
+    get formatedBirthdate() {
+        return formatDate(
```
Comment (cvinh) [inline pos_partner_birthdate/static/src/js/PartnerLine.esm.js:9]: formatDate should not be used as birthdate does not depend on tz
Steps to reproduce : setup your tz to gmt-10 nd the birthdate changes to -1day

File: pos_partner_birthdate/static/src/js/PosDB.esm.js (hunk 1 above)
Comment (etobella) [reply pos_partner_birthdate/static/src/js/PosDB.esm.js]: Log removed :wink:

**Review by cvinh — CHANGES_REQUESTED** [review]

File: pos_partner_birthdate/static/src/js/PartnerLine.esm.js (hunk 3 above)
Comment (etobella) [reply pos_partner_birthdate/static/src/js/PartnerLine.esm.js:9]: You are right, I was able to reproduce the error forcing the zone of the navigator. I fixed it forcing UTC timezone :wink:

File: pos_partner_birthdate/static/src/js/PosDB.esm.js (hunk 2 above)
Comment (etobella) [reply pos_partner_birthdate/static/src/js/PosDB.esm.js]: The problem with this configuration is that we do not have this settings. It is not even accepted by Odoo, making it hard to manage :sob: 

Odoo tries to parse the search parameters to datetime in order to do that, however, it only works on you locale standard format

Comment (cvinh) [reply pos_partner_birthdate/static/src/js/PosDB.esm.js]: The user does not want to enter the '/' (same as in the backend)
Maybe you can add the '/' with some tricky js or use some code from web... ?

Comment (etobella) [reply pos_partner_birthdate/static/src/js/PosDB.esm.js]: Ok, I can add a second filter with cleaned data (removing `.`, `-`and `/`)

Comment (cvinh) [reply pos_partner_birthdate/static/src/js/PosDB.esm.js]: It works like a charm

**Review by cvinh — APPROVED** [review]

**Review by ecino — APPROVED** [review]

**Pull request merged**